torch/jit/__init__.py

```diff
@@ -106,13 +105,36 @@ def compile(arg=None, **kwargs):
     function must declare *all* Variable inputs in its arguments.
 
         >>> @jit.compile
-        >>> def f(x);
+        >>> def f(x):
         >>>     return x * 2
     """
     def _compile(arg):
         if inspect.isclass(arg):
-            if issubclass(arg, _CompiledMixin):
-                raise TypeError("Cannot compile a model class that already is compiled")
+            class CompiledModuleMeta(type):
+                def __call__(cls, *args, **kwargs):
+                    # NOTE: this is called whenever an instance of this class is created
+                    # The super call below will call __new__ and __init__, and we will
+                    # patch things later.
+                    try:
+                        obj = super(CompiledModuleMeta, cls).__call__(*args, **kwargs)
+                    except TypeError as e:
+                        # If this fails here, the user probably didn't use this as a class decorator
+                        if "super" in str(e):
+                            raise_from(TypeError("torch.jit.compile must be used as a class decorator; "
+                                                "using it on an already defined class is not valid."
+                                                "\n\nOriginal error: {}".format(str(e))), e)
+                        else:
+                            raise
+
+                    meth = obj.forward
```

Comment:
used?

Reply:
That looks like a leftover from a few quick experiments I ran. Will remove tomorrow